### ISSUE TITLE
Automated test suite

### ISSUE THREAD

**Issue by ppaalanen** [contactshashanksharma/libedid#4]
One of the major points I wanted from a libedid project is that it has its own CI with an extensive EDID database to test against. The edid-decode project already has a database I believe.

How to make that happen, I do not know.

Starting with a manually executed test suite would be a good start. CI integration might be best done after there is consensus where this project should live. If this project is taken under freedesktop.org, which I would hope, then you'd be using Gitlab CI which is different from Github.

Obviously this can't be started until we have some API agreed upon.

**Comment by contactshashanksharma** [comment]
Yes, I also think manually executed test suit would be a good start, once we finish with some of the high level APIs. But I will need some help with the CI integration or automated suits (or I will have to study a lot of stuff here :) ).

Let's talk more on the project's location, along with the API discussion.

**Comment by ppaalanen** [comment]
If you go with Meson, then `meson test` is what we'd use both manually and in CI to run the tests. In Meson you define a test with a program to run, and you have the choice of using e.g. process exit code or TAP for reporting results. Then `meson test` runs all defined tests and shows a summary.

Another matter is, if one needs to e.g. download an EDID database manually for that to work. Manual download with instructions is a good first step. What database to choose and how to make use of it is the part I don't know. Also if you want to use some ready-made testing framework, that is up to you.

OTOH, I can help with Meson and Gitlab CI once we're that far.

**Comment by contactshashanksharma** [comment]
Sounds like a plan, let me start with the high level API, once we have something concrete there, we can prepare a focused plan on how to move to Mason
